ci: Fix mac names, bump builds to bigger runner

seemethere opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #81882
* __->__ #81874

Mac names were inconsistent so I changed them to be more reflective of
what they're actually testing. As well wanted to update the builders for
the build portions to run on more powerful runners to reduce tts

Some data:
* Cuts build time by 2/3 for `arm64` cross builds:
  * [Trunk](https://github.com/pytorch/pytorch/runs/7447691552?check_suite_focus=true): 1h 22min
  * [This PR](https://github.com/pytorch/pytorch/runs/7449648493?check_suite_focus=true): 30 min
* Similar speed ups for `x86`:
  * [Trunk](https://github.com/pytorch/pytorch/runs/7447691171?check_suite_focus=true): 1h 47min
  * [This PR](https://github.com/pytorch/pytorch/runs/7449647307?check_suite_focus=true): 41min

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

seemethere committed:
ci: Fix mac names, bump builds to bigger runner

Mac names were inconsistent so I changed them to be more reflective of
what they're actually testing. As well wanted to update the builders for
the build portions to run on more powerful runners to reduce tts

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

[ghstack-poisoned]

albanD commented on .github/workflows/trunk.yml:
```diff
@@ -160,36 +160,36 @@ jobs:
+    needs: macos-12-py3-arm64-build
     with:
-      build-environment: macos-10-15-py3-arm64
+      build-environment: macos-12-py3-arm64
```
For consistency with the other jobs here, do we want to specify the runner type here instead of `_mac-test-arm64.yml`?

seemethere replied:
Perhaps leaving a comment here might suffice

albanD replied:
No need to do this here for sure.
I was checking the runners for all of these and realized this one is special!

albanD reviewed (approved):
The change looks ok.
Hopefully these names are not hardcoded anywhere else!

seemethere committed:
Update on "ci: Fix mac names, bump builds to bigger runner"


Mac names were inconsistent so I changed them to be more reflective of
what they're actually testing. As well wanted to update the builders for
the build portions to run on more powerful runners to reduce tts

Signed-off-by: Eli Uriegas <eliuriegasfb.com>

[ghstack-poisoned]

atalman reviewed (approved):
LGTM
